### ISSUE TITLE
Distinct filter does not filter out results in query

### ISSUE THREAD

**Issue by tlammer** [specify/specify7#331]
Create a query (I just added the Sex field and said not empty), check the Distinct box, and run the query.  The results should be just the unique values for that field.  Instead, you get every record result that is not empty (following the condition placed on Sex).

**Comment by tlammer** [comment]
commenting on this because it is still an issue and UMich has noticed.

**Comment by maxxxxxdlp** [comment]
Still reproducable

**Comment by maxxxxxdlp** [comment]
Fixed by not returning record IDs if the query is set to "Distinct"
This means that the records from the query results can not be opened

I checked and it seems like there are no back-end distinct queries and there was only on front-end distinct query (the lifemapper map. I made that query not be distinct). Thus, this change should not break dependent components (query boxes, express search, etc...)

---
@benanhalt the only thing I still do not understand is what does "this.forceResourceLoad" here mean? https://github.com/specify/specify7/blob/237b398d40a65744c7862fd8e7726e6159170669/specifyweb/frontend/js_src/lib/queryresults.js#L228
This part is probably going to throw an error if "Distinct" filter is set
